### PR TITLE
JNG-3336_Self_not_allowed_in_lambda

### PR DESCRIPTION
**Pull request by gprivi**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-3336" title="JNG-3336" target="_blank">Sub-taskJNG-3336</a>  karaf-features-antlr
  </td></table>
  <br />
 

JNG-3336_Self_not_allowed_in_lambda

### CODE REVIEW

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/scoping/JslDslScopeProvider.xtend [hunk 1]
```diff
@@ -161,7 +162,8 @@ class JslDslScopeProvider extends AbstractJslDslScopeProvider {
 				FunctionDeclaration: return true
 				LambdaDeclaration: return true
 				ErrorDeclaration: return true
-				LambdaVariable: return EcoreUtil2.getAllContainers(context).exists[c | c.isEqual(obj.eContainer)] || context.isEqual(obj.eContainer)
+//				LambdaVariable: return EcoreUtil2.getAllContainers(context).exists[c | c.isEqual(obj.eContainer)] || context.isEqual(obj.eContainer)
```
Comment (robertcsakany) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/scoping/JslDslScopeProvider.xtend]: Please remove it

**Review by robertcsakany — CHANGES_REQUESTED** [review]
Plase add tests for lambda self validation.

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/scoping/JslDslScopeProvider.xtend (hunk 1 above)
Comment (gprivi) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/scoping/JslDslScopeProvider.xtend]: Done

**Comment by gprivi** [comment]
> Plase add tests for lambda self validation.

Done

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**